### PR TITLE
feat(app): create Banner component and add throughout module cards

### LINKED ISSUE #9290: H/S Module Card: Hot to touch alert
# Overview

When the H/S is above a temp threshold of 48.9C/120F, a Hot to Touch alert will be displayed on the H/S module card

# Implementation Details
- When the Heater shaker is above temp threshold of 48.9C/120F, an alert will display on the module card indicating the module is hot to touch (this matches with module LED behavior)
- The alert should be styled and laid out as specified in the design

# Design Links
- https://www.figma.com/file/695ogls23YsQYImRcGYDE1/Heater-Shaker?node-id=740%3A31018

# Acceptance Criteria
- [ ] When the Heater shaker is above temp threshold of 48.9C/120F, an alert will display on the module card indicating the module is hot to touch (this matches with module LED behavior)
- [ ] The alert should be styled and laid out as specified in the design

### PR DESCRIPTION
**Pull request by jerader**
closes #9290 #9545

# Overview

This PR creates the new Alert Item (called Banner) and implements it in both the module card if there is a firmware update and the heater shaker module card if the module is too hot to touch. Also, this updates the usage of `AlertItem` in the About Module Slideouts to `Banner` and updates the button styling.

Screen Shot 2022-03-22 at 10 00 27 AM

Screen Shot 2022-03-22 at 10 01 14 AM

Screen Shot 2022-03-22 at 10 01 03 AM

# Changelog

- creates `Banner` and deprecates the `AlertItem` component
- creates a story and a test for `Banner`
- updates the usage of `AlertItem` in the `AboutModule` slideout and updates the button to match new designs
- adds `Banner` to module cards if the H-S is too hot to touch
- adds `Banner` to module cards if the module has an update available

# Review requests

- review above images, there should be a `banner` the matches figma and the banners should display when they are supposed to

Note: there is still a few TODOs to wire up the `Firmware update` buttons. Still not sure where they're supposed to lead you to. But there should be a ticket to address those

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/ModuleCard/__tests__/AboutModuleSlideout.test.tsx [hunk 1]
```diff
@@ -141,6 +141,8 @@ describe('AboutModuleSlideout', () => {
     const viewUpdate = getByRole('button', { name: 'View Update' })
     fireEvent.click(viewUpdate)
     expect(viewUpdate).toBeEnabled()
+    const exit = getByLabelText('close_icon')
+    fireEvent.click(exit)
```
Comment (shlokamin) [inline app/src/organisms/Devices/ModuleCard/__tests__/AboutModuleSlideout.test.tsx:145]: we should also verify `onCloseClick` gets called when clicking the close button

File: app/src/atoms/Banner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  & a {
+    text-decoration: underline;
+    color: inherit;
+  }
+
+  &:empty {
+    padding: 0;
+  }
+`
+export function Banner(props: BannerProps): JSX.Element {
```
Comment (shlokamin) [inline app/src/atoms/Banner/index.tsx:71]: this is great! i don't think it's an atom though (atoms are our smallest units that shouldnt be able to be split up anymore). i'd say this is probably a molecule?

File: app/src/organisms/Devices/ModuleCard/__tests__/ModuleCard.test.tsx [hunk 3]
```diff
@@ -148,4 +178,17 @@ describe('ModuleCard', () => {
...
+      module: mockHotHeaterShaker,
+    })
+    getByText('mock banner')
+  })
+  it('renders information for a magnetic module when an update is available so update banner renders', () => {
+    const { getByText } = render({
+      module: mockMagneticModuleHub,
+    })
+    getByText('mock banner')
+  })
```
Comment (shlokamin) [inline app/src/organisms/Devices/ModuleCard/__tests__/ModuleCard.test.tsx]: these two tests seem like they're trying to make sure that two different banners get rendered, but we aren't actually asserting any difference. we should either...

1. make sure the banner component is called with the expected props before rendering OR
2. dont mock the banner component and find by text

File: app/src/organisms/Devices/ModuleCard/index.tsx [hunk 4]
```diff
@@ -56,11 +60,18 @@ export const ModuleCard = (props: ModuleCardProps): JSX.Element | null => {
...
+  const hotToTouch: IconProps = { name: 'hot-to-touch' }
 
   const moduleOverflowWrapperRef = useOnClickOutside({
     onClickOutside: () => setShowOverflowMenu(false),
   }) as React.RefObject<HTMLDivElement>
 
+  const tooHot =
+    module.model === 'heaterShakerModuleV1' &&
+    module.data.currentTemp != null &&
+    module.data.currentTemp > 48.9
```
Comment (shlokamin) [inline app/src/organisms/Devices/ModuleCard/index.tsx]: can this be moved into a constant in shared data?

File: app/src/organisms/Devices/ModuleCard/index.tsx [hunk 5]
```diff
@@ -56,11 +60,18 @@ export const ModuleCard = (props: ModuleCardProps): JSX.Element | null => {
...
   const [hasSecondary, setHasSecondary] = React.useState(false)
   const [showAboutModule, setShowAboutModule] = React.useState(false)
+  const [showBanner, setShowBanner] = React.useState<boolean>(true)
+  const hotToTouch: IconProps = { name: 'hot-to-touch' }
 
   const moduleOverflowWrapperRef = useOnClickOutside({
     onClickOutside: () => setShowOverflowMenu(false),
   }) as React.RefObject<HTMLDivElement>
 
+  const tooHot =
```
Comment (shlokamin) [inline app/src/organisms/Devices/ModuleCard/index.tsx]: can this be prefixed with an `is` to indicate it's a boolean?

File: components/src/icons/icon-data.ts [hunk 6]
```diff
@@ -429,4 +429,9 @@ export const ICON_DATA_BY_NAME = {
     path:
       'M12 0C9.61305 0 7.32387 0.948211 5.63604 2.63604C3.94821 4.32387 3 6.61305 3 9H0L3.89 12.89L3.96 13.03L8 9H5C5 7.14348 5.7375 5.36301 7.05025 4.05025C8.36301 2.7375 10.1435 2 12 2C13.8565 2 15.637 2.7375 16.9497 4.05025C18.2625 5.36301 19 7.14348 19 9C19 10.8565 18.2625 12.637 16.9497 13.9497C15.637 15.2625 13.8565 16 12 16C10.07 16 8.32 15.21 7.06 13.94L5.64 15.36C7.27 17 9.5 18 12 18C14.3869 18 16.6761 17.0518 18.364 15.364C20.0518 13.6761 21 11.3869 21 9C21 6.61305 20.0518 4.32387 18.364 2.63604C16.6761 0.948211 14.3869 0 12 0Z',
   },
+  'hot-to-touch': {
```
Comment (shlokamin) [inline components/src/icons/icon-data.ts]: is this an opentrons specific icon? if so i think this should be prefixed with `ot-`, but tbh not sure if its generic

**Review by shlokamin — APPROVED** [review]
left some nits but changes LGTM!

**Review by Kadee80 — COMMENTED** [review]
just double checking designs. looks right in the banner itself in the screenshot the X is taking on color instead of being COLORS.darkBlack 

functionality looks dope though. i'm down to do a pas in my upcoming PR for any nits ad picks.

P.S. thanks for adding the icon variability

File: components/src/icons/icon-data.ts (hunk 6 above)
Comment (Kadee80) [reply components/src/icons/icon-data.ts]: pretty sure thats the best name of an icon we have but +1 for` ot-`

File: app/src/atoms/Banner/index.tsx (hunk 2 above)
Comment (jerader) [reply app/src/atoms/Banner/index.tsx:71]: I thought @b-cooper said during standup yesterday that `banner` is an atom?

**Comment by jerader** [comment]
> just double checking designs. looks right in the banner itself in the screenshot the X is taking on color instead of being COLORS.darkBlack

@Kadee80, totally forgot to update the pics in my PR but I did change the X back to black. I think I was looking at an out of date figma when I first made it. 

Screen Shot 2022-03-23 at 10 18 06 AM

**Pull request merged**